### PR TITLE
Added support of the SingleStoreDB

### PR DESCRIPTION
**Pull request by AdalbertMemSQL**
Added support of the SingleStore database using [SingleStore Driver for Laravel](https://github.com/singlestore-labs/singlestoredb-laravel-driver/)

### CODE REVIEW

**Comment by heychazza** [comment]
Thank you so much for adding this PR, it’s going to be superb to use CTE’s with SingleStore nicely (right now I use raw SQL which isn’t ideal).

@staudenmeir any chance this could be reviewed and released to Composer?

File: composer.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
     ],
     "require": {
         "php": "^8.1",
-        "illuminate/database": "^10.0"
+        "illuminate/database": "^10.0",
+        "singlestoredb/singlestoredb-laravel": "^1.5.1"
```
Comment (staudenmeir) [inline composer.json:14]: Please move the package to the `require-dev` section. It shouldn't be included but installed separately.

File: composer.json [hunk 2]
```diff
@@ -30,7 +31,8 @@
     "extra": {
         "laravel": {
             "providers": [
-                "Staudenmeir\\LaravelCte\\DatabaseServiceProvider"
+                "Staudenmeir\\LaravelCte\\DatabaseServiceProvider",
+                "SingleStore\\Laravel\\SingleStoreProvider"
```
Comment (staudenmeir) [inline composer.json:35]: Why did you add the service provider here?

File: src/Connectors/ConnectionFactory.php [hunk 3]
```diff
@@ -26,7 +27,7 @@ class ConnectionFactory extends Base
      */
     protected function createConnection($driver, $connection, $database, $prefix = '', array $config = [])
     {
-        if ($resolver = Connection::getResolver($driver)) {
+        if ($driver != 'singlestore' && $resolver = Connection::getResolver($driver)) {
```
Comment (staudenmeir) [inline src/Connectors/ConnectionFactory.php:30]: Why is this necessary?

Comment (staudenmeir) [reply src/Connectors/ConnectionFactory.php:30]: @AdalbertMemSQL

Comment (AdalbertMemSQL) [reply src/Connectors/ConnectionFactory.php:30]: Sorry, I forgot to send the answer

Comment (staudenmeir) [reply src/Connectors/ConnectionFactory.php:30]: Thanks

**Comment by staudenmeir** [comment]
Thanks! I released a new version.
